src/Voron/Impl/Paging/AbstractPager.cs

```diff
@@ -303,14 +303,21 @@ public int NodeMaxSize
                     goto AlreadyDisposed;
             }
 
-            tx?.EnsurePagerStateReference(state);
+            tx?.EnsurePagerStateReference(ref state);
```

Comment:
This is the most important change - passing `state` using `ref` so we can be sure that below `state.MapBase ...` usage is valid. The issue was that `state` could be disposed during `EnsurePagerStateReference` but we were still using it below. See the test which proves that can be an issue.

That could cause AVE (like here - https://issues.hibernatingrhinos.com/issue/RavenDB-16128) and also `ObjectDisposedException` (original problem that caused the investigation - https://issues.hibernatingrhinos.com/issue/RavenDB-16507) if such a state was passed to the ctor of LowLevelTransaction used for the transaction cloning feature.